Make it possible to hide adsb vehicle

Fredrikbjornland opened the pull request.

kibidev reviewed (changes requested):
Will look at bit more, but some initial findings.

Main thing: segfault:
[Screencast from 23. okt. 2024 kl. 13.10 +0200.webm](https://github.com/user-attachments/assets/3416983c-c86c-48cc-81c3-f77559fc89f6)

I also think it is not needed with confirm to show entries again (but check with OPS).
Perhaps also a "show all" option.

Fredrikbjornland commented on src/FlightDisplay/FlyViewMap.qml:
```diff
@@ -268,6 +268,55 @@ FlightMap {
     }
 
+
+    // Handle guided mode clicks
```
Yeah I agree, but I spent quite some time trying different things and I did not get it to work, even though it works for waypoints. Will move ADSB code below

Fredrikbjornland commented:
> Will look at bit more, but some initial findings.
> 
> Main thing: segfault: [Screencast from 23. okt. 2024 kl. 13.10 +0200.webm](https://github.com/user-attachments/assets/3416983c-c86c-48cc-81c3-f77559fc89f6)
> 
> I also think it is not needed with confirm to show entries again (but check with OPS). Perhaps also a "show all" option.

Fixed segfault by emitting hasHiddenADSBVehicleChanged in _cleanupStaleVehicles().
Removed confirm when showing entry again
Added "Show all"-button

kibidev reviewed (changes requested):
Very close to mergeable now, I think

Fredrikbjornland commented on src/FlightMap/MapItems/VehicleMapItem.qml:
```diff
@@ -107,6 +112,33 @@ MapQuickItem {
+            MouseArea {
+                anchors.fill: parent
+                enabled:      _adsbVehicle
+                cursorShape:  _adsbVehicle ?  Qt.PointingHandCursor : Qt.ArrowCursor
```
I agree that the cursor is ugly, but I think it is good UX to change it regardless. Changing the cursor tells the user the vehicle can be clicked, and it could be a nice reminder to see that the cursor changes, in case they forgot you can hide adsb vehicles. But you are right in that cursor change is not really used in QGC, so I am open to removing it if you think we should?

kibidev replied:
If it was consistently used, it could be ok to change it (regardless of it not being very pretty), but I think we should not add special cursor behaviour for this alone.

Fredrikbjornland commented on src/ADSB/ADSBVehicleManager.cc:
```diff
@@ -191,3 +191,24 @@ void ADSBTCPLink::_parseLine(const QString& line)
+        }
+    } else {
+        qCDebug(ADSBVehicleManagerLog) << "ADSBVehicleManager: ICAO address not found";
+    }
```
Yeah you were right, changed it to your suggestion
